converters/toZigbee.js

```diff
@@ -46,6 +46,25 @@ const converters = {
             entity.commandResponse('ssIasAce', 'panelStatusChanged', payload);
         },
     },
+    set_status: {
```

Comment:
I think https://github.com/Koenkk/zigbee-herdsman-converters/blob/2ff4fc1964cbcc880eb11782b051d8014b58cb21/converters/toZigbee.js#L31 can be used instead.

Reply:
@Koenkk Tried it, doesn't do anything on the panel.
And if i look at this code it has more functionality?

@koenk @dettrick 
 utils.getKeyByValue isn't doesn't exist anymore as far as i can see. You must use utils.getKey
 
Also this line is incorrect ( as far as my knowledge goes)
const panelStatus = utils.getKeyByValue(constants.panelStatus, value.panel_status, undefined, Number);
value.panel_status is never set

Reply:
@Koenk @dettrick 
Still getting this error:
Zigbee2MQTT:error 2021-01-21 10:36:50: Failed to call 'EntityPublish' 'onMQTTMessage' (TypeError: Cannot read property 'key' of undefined
    at /app/lib/extension/publish.js:185:56
    at Array.find (<anonymous>)
    at EntityPublish.onMQTTMessage (/app/lib/extension/publish.js:185:42)
    at Controller.callExtensionMethod (/app/lib/controller.js:382:44)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5))
  
  when i send to MQTT topic  /zigbee2mqtt/[FRIENDLY_NAME]/set
  Message:
  { "set_status": { "panelstatus":"arming_away", "secondsremain":0, "audiblenotif":0, "alarmstatus":0
} }

Reply:
You are missing the underscore for each of the key's in your set_status object. That is why it cant find the keys because they dont match what the code is looking for. Your MQTT message should be:

{ "set_status": { "panel_status":"arming_away", "seconds_remain":0, "audible_notif":0, "alarm_status":0
} }

Reply:
Changed it, but still getting the same error. ( NOTE: I've used my changed code )

Reply:
Also just tried your code. Also not working

Reply:
Ok, does arm_mode work?

Reply:
That seems to work although i can't see anywhere if it's armed/done the job:
This is in the log:
debug Publishing 'set' 'arm_mode' to '0x000d6f0013e97ce5'
debug Saving state to file /share/zigbee2mqtt/state.json

So i guess that works

Used this Json:
{
    "arm_mode": {
      "mode": "arm_all_zones"
    }
}

Reply:
If i copy the exact same code as 'arm_code' and rename it to 'set_status' restart zigbee2mqtt and i use:
{
"set_status": {
"mode": "arm_all_zones"
}
}
I get the error too:
Failed to call 'EntityPublish' 'onMQTTMessage' (TypeError: Cannot read property 'key' of undefined at /app/lib/extension/publish.js:185:56 at Array.find (<anonymous>) at EntityPublish.onMQTTMessage (/app/lib/extension/publish.js:185:42) at Controller.callExtensionMethod (/app/lib/controller.js:382:44) at processTicksAndRejections (internal/process/task_queues.js:97:5))

Reply:
In your MQTT, have you tried using a different panel_status such as 'disarmed' or 'armed_away'?

Reply:
Yep, tried your code...my code...and the renamed 'arm_mode' code ..they all give me:
Zigbee2MQTT:error 2021-01-21 13:51:24: Failed to call 'EntityPublish' 'onMQTTMessage' (TypeError: Cannot read property 'key' of undefined
    at /app/lib/extension/publish.js:185:56
    at Array.find (<anonymous>)
    at EntityPublish.onMQTTMessage (/app/lib/extension/publish.js:185:42)
    at Controller.callExtensionMethod (/app/lib/controller.js:382:44)
    at processTicksAndRejections (internal/process/task_queues.js:97:5))

Reply:
Found the problem! The toZigbee,js and fromZigbee.js aren't loaded in the dev! Don't know how to force that!

Reply:
So does `arm_mode` work now? Reading the code this `set_status` is almost identical to `arm_mode`.

Reply:
@Koenkk arm_mode works but the indicator lights and beeps/buzzer won't work with arm_mode. That's why we need set_status.
So yes it almost looks the same..but it isn't :)

The problem is that i can't test it because i can override the devices.js  but it WON'T load the /converters.
Allready tried tricking into custom converters..then zigbee2mqtt won't start.
Followed this: https://github.com/danielwelch/hassio-zigbee2mqtt#adding-support-for-new-devices

I want to help with the code and to get this merged asap but it's impossible if i can't load the new/edited converters

Reply:
> So yes it almost looks the same..but it isn't :)

What's the difference then?

Reply:
It sets the status of the panel itself. The led's of the panel and the buzzer/beeper. Something arm_mode doesn't do.

 I can disarm the panel with arm_mode  but when i arm it again the disarm led doesn't turn off and the led that needs ro turn on doesn't turn on. Also buzzer/beeper doesn't do anything with arm_ mode

I hope this explanation is clear enough. If i could get the dev mode working right i could send you a video but i can't get it to work right so i can't.
Maybe dettrick can of you need /want it

Reply:
For the Xfinity keypad arm_status does nothing to be honest, at least as far user functionality is concerned. On other keypads however it may do more an without testing all keypads, we can't know. Given that we haven't touched the "arm_status"  code as part of this update, let's leave it as it is.

Reply:
This set_status looks like an almost duplicate of the arm_mode. So this just looks like duplicate code. Please explain me what line of code makes the difference here.

Reply:
@Koenkk I've installed a dev version on a Ubuntu machine with an old CC2531 coordinator so i can help @dettrick :) :

The code in ARM_MODE:
"const panelStatus = mode !== 0 && mode !== 4 ? 0x80: 0x00;"
Renders the Xfinity display status to nonreponsive.

As soon as you change the line of ARM_MODE to:
"const panelStatus = mode;"
It WILL work BUT then you have to extend the 'armMode' in  constant.js to:
const armMode = {
    0: 'disarm', // NOTE IS SAME AS "DISARMED" ON XFINITY
    1: 'arm_day_zones', // NOTE IS SAME AS "ARMED_STAY" ON XFINITY  ( FIRST LED )
    2: 'arm_night_zones',// NOTE IS SAME AS "ARMED_NIGHT" ON XFINITY  ( THIRD LED )
    3: 'arm_all_zones',// NOTE IS SAME AS "ARMED_AWAY" ON XFINITY  ( SECOND LED )
    4: 'invalid_code', 
    5: 'exit_delay', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
    6: 'entry_delay', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
    7: 'not_ready', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
    8: 'in_alarm', //NEW  for xfinity xhk1-eu --> Doesn't seem to do anything ? @dettrick
    9: 'arming_stay', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
    10: 'arming_night', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
    11: 'arming_away', //NEW  for xfinity xhk1-eu --> Led +beep on XHK1-EU Panel
};

So if there is a way to get rid of the mode !== 4 ? 0x80: 0x00;  and extend the armMode in constants.js the Xfinity XHK1-EU panel could use the arm_mode

So @koenk if you have any ideas on how to get rid of that..please share :)

Reply:
Thanks @Multipage !

Regarding panelStatus, changing `const panelStatus = mode !== 0 && mode !== 4 ? 0x80: 0x00;` to

```js
let panelStatus = mode;
if (meta.mapped.model === '3400-D') {
    panelStatus = mode !== 0 && mode !== 4 ? 0x80: 0x00;
}
```

Regarding arm mode, add the new entries (but don't change existing ones because this is a breaking change).

Reply:
Thanks @Koenkk . Just tried it and it works.
@dettrick : Could you change the toZigbee.js and the constants.js 

toZigbee.js :
Delete the **'set_status'.**
And edit the **arm_mode**
**Change this line:** 
`const panelStatus = mode !== 0 && mode !== 4 ? 0x80: 0x00;`

**to:**
```
let panelStatus = mode;
if (meta.mapped.model === '3400-D') {
    panelStatus = mode !== 0 && mode !== 4 ? 0x80: 0x00;
}
```

**constants.js:**
Extend the constant **armMode** to:
```
const armMode = {
  0: 'disarm', 
  1: 'arm_day_zones', 
  2: 'arm_night_zones',
  3: 'arm_all_zones',
  4: 'invalid_code',
  5: 'exit_delay', 
  6: 'entry_delay',
  7: 'not_ready',
  8: 'in_alarm',
  9: 'arming_stay',
  10: 'arming_night',
  11: 'arming_away',
};
```
ps: I've deleted some replies.

Reply:
All done

Reply:
Hi @dettrick. Just a tiny thing left.

In constants.js remove the
```
const panelStatus = {
    0: 'disarmed',
    1: 'armed_stay',
    2: 'armed_night',
    3: 'armed_away',
    4: 'exit_delay',
    5: 'entry_delay',
    6: 'not_ready',
    7: 'in_alarm',
    8: 'arming_stay',
    9: 'arming_night',
    10: 'arming_away',
};
```

And
`panelStatus,`
from module.exports

Then we are all set for merging i think?
Thanks for your work @dettrick

Reply:
No problemo. All done